unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quests/QuestLogComponentView.cs

```diff
@@ -0,0 +1,226 @@
+            completedQuests = new ();
+            questsPool = new UnityObjectPool<ActiveQuestComponentView>(activeQuestPrefab, activeQuestsContainer, actionOnDestroy: x => x.Hide());
+            questsPool.Prewarm(MAX_QUESTS_COUNT);
+
+            InitialiseSectionSelector();
+
+            questDetailsComponentView.OnJumpIn += (coords) => OnJumpIn?.Invoke(coords);
+            questDetailsComponentView.OnPinChange += (questId, isPinned) => OnPinChange?.Invoke(questId, isPinned);
+            questDetailsComponentView.OnQuestAbandon += (questId) => OnQuestAbandon?.Invoke(questId);
+
+            emptyState.SetActive(true);
+        }
+
+        private void InitialiseSectionSelector()
+        {
+            sectionSelector.Awake();
+            sectionSelector.GetSection(IN_PROGRESS_SECTION_INDEX).onSelect.RemoveAllListeners();
+            sectionSelector.GetSection(COMPLETED_SECTION_INDEX).onSelect.RemoveAllListeners();
+            sectionSelector.GetSection(IN_PROGRESS_SECTION_INDEX).onSelect.AddListener((isSelected) =>
+            {
+                if (!isSelected) return;
+
+                headerText.text = IN_PROGRESS_TITLE;
+                ShowActiveOrCompletedQuests(true);
+                if (activeQuests.Count == 0)
+                {
+                    emptyState.SetActive(true);
+                    emptyActiveState.SetActive(true);
+                    emptyCompletedState.SetActive(false);
+                }
+                else
+                {
+                    emptyState.SetActive(false);
+                }
+            });
+            sectionSelector.GetSection(COMPLETED_SECTION_INDEX).onSelect.AddListener((isSelected) =>
+            {
+                if (!isSelected) return;
+
+                headerText.text = COMPLETED_TITLE;
+                ShowActiveOrCompletedQuests(false);
+                if (completedQuests.Count == 0)
+                {
+                    emptyState.SetActive(true);
+                    emptyActiveState.SetActive(false);
+                    emptyCompletedState.SetActive(true);
+                }
+                else
+                {
+                    emptyState.SetActive(false);
+                }
+            });
+        }
+
+        private void ShowActiveOrCompletedQuests(bool active)
+        {
+            foreach (var activeQuest in activeQuests.Values)
+            {
+                activeQuest.gameObject.SetActive(active);
+            }
+            foreach (var completedQuest in completedQuests.Values)
+            {
+                completedQuest.gameObject.SetActive(!active);
+            }
+        }
+
+        public void AddActiveQuest(QuestDetailsComponentModel activeQuest)
+        {
+            emptyState.SetActive(false);
+
+            if (!activeQuests.ContainsKey(activeQuest.questId))
+                activeQuests.Add(activeQuest.questId, questsPool.Get());
+
+            activeQuests[activeQuest.questId].OnActiveQuestSelected -= SelectedQuest;
```

Comment:
`activeQuests[activeQuest.questId]` is used 3 times, I would extract it into a local variable

Reply:
Same case at `AddCompletedQuest`